### PR TITLE
libplacebo 4.208.0 (new formula)

### PR DESCRIPTION
**Pull request by rcombs**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by rcombs** [comment]
This doesn't build on very old gcc versions (≤5) due to lack of support for enumerator attributes. I filed an MR against upstream (https://code.videolan.org/videolan/libplacebo/-/merge_requests/283), but I'm not entirely sure anyone will care for something that obsolete; added a `fails_with` for now.

File: Formula/libplacebo.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    url "https://files.pythonhosted.org/packages/1d/97/2288fe498044284f39ab8950703e88abbac2abbdf65524d576157af70556/MarkupSafe-2.1.1.tar.gz"
+    sha256 "7f91197cc9e48f989d12e4e6fbc46495c446636dfc81b9ccf50bb0ec74b91d4b"
+  end
+
+  # We have to grab meson ourselves rather than use the brew package,
+  # since we're going to need Mako and MarkupSafe to be installed in its venv.
+  resource "meson" do
+    url "https://files.pythonhosted.org/packages/c5/03/fa2ec33f079e9b31bbd2154257cdfe51de3f085139e75aa945fd2ecdfe86/meson-0.63.0.tar.gz"
+    sha256 "3b51d451744c2bc71838524ec8d96cd4f8c4793d5b8d5d0d0a9c8a4f7c94cd6f"
+  end
```
Comment (carlocab) [inline Formula/libplacebo.rb:38]: Why not use `PYTHONPATH` instead?

File: Formula/libplacebo.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  def install
+    venv = virtualenv_create(buildpath/"venv", "python3")
+    venv.pip_install resources
+    ENV.prepend_path "PATH", buildpath/"venv/bin"
+
+    mkdir "build" do
+      system "meson", "-Dvulkan-registry=#{Formula["vulkan-headers"].share}/vulkan/registry/vk.xml", *std_meson_args
+      system "ninja", "-v"
+      system "ninja", "install", "-v"
+    end
```
Comment (carlocab) [inline Formula/libplacebo.rb:49]:
```suggestion
    system "meson", "setup", "build", "-Dvulkan-registry=#{Formula["vulkan-headers"].share}/vulkan/registry/vk.xml", *std_meson_args
    system "meson", "compile", "-C", "build", "--verbose"
    system "meson", "install", "-C", "build"
```

File: Formula/libplacebo.rb (hunk 1 above)
Comment (rcombs) [reply Formula/libplacebo.rb:38]: I'm not sure how to do that for this.

File: Formula/libplacebo.rb (hunk 2 above)
Comment (rcombs) [reply Formula/libplacebo.rb]: Done.

File: Formula/libplacebo.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  # since we're going to need Mako and MarkupSafe to be installed in its venv.
+  resource "meson" do
+    url "https://files.pythonhosted.org/packages/c5/03/fa2ec33f079e9b31bbd2154257cdfe51de3f085139e75aa945fd2ecdfe86/meson-0.63.0.tar.gz"
+    sha256 "3b51d451744c2bc71838524ec8d96cd4f8c4793d5b8d5d0d0a9c8a4f7c94cd6f"
+  end
+
+  def install
+    venv = virtualenv_create(buildpath/"venv", "python3")
+    venv.pip_install resources
+    ENV.prepend_path "PATH", buildpath/"venv/bin"
```
Comment (carlocab) [inline Formula/libplacebo.rb:43]:
```suggestion
  def install
    python = "python3.10"
    venv_root = buildpath/"venv"
    venv = virtualenv_create(venv_root, python)
    venv.pip_install resources
    ENV.prepend_path "PYTHONPATH", venv_root/Language::Python.site_packages(python)
```

File: Formula/libplacebo.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  include Language::Python::Virtualenv
+
+  desc "Reusable library for GPU-accelerated image/video processing primitives"
+  homepage "https://code.videolan.org/videolan/libplacebo"
+  url "https://code.videolan.org/videolan/libplacebo/-/archive/v4.208.0/libplacebo-v4.208.0.tar.bz2"
+  sha256 "c89a80655ab375e4809415bb597c638607fc150fa6f6bb830dd502fec7f0ba95"
+  license "LGPL-2.1-or-later"
+  head "https://code.videolan.org/videolan/libplacebo.git", branch: "master"
+
+  depends_on "ninja" => :build
```
Comment (carlocab) [inline Formula/libplacebo.rb:11]:
```suggestion
  depends_on "meson" => :build
  depends_on "ninja" => :build
```

File: Formula/libplacebo.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    sha256 "3b51d451744c2bc71838524ec8d96cd4f8c4793d5b8d5d0d0a9c8a4f7c94cd6f"
+  end
+
+  def install
+    venv = virtualenv_create(buildpath/"venv", "python3")
+    venv.pip_install resources
+    ENV.prepend_path "PATH", buildpath/"venv/bin"
+
+    system "meson", "setup", "build", "-Dvulkan-registry=#{Formula["vulkan-headers"].share}/vulkan/registry/vk.xml",
+           *std_meson_args
```
Comment (carlocab) [inline Formula/libplacebo.rb:46]:
```suggestion
    system "meson", "setup", "build",
                    "-Dvulkan-registry=#{Formula["vulkan-headers"].share}/vulkan/registry/vk.xml",
                    *std_meson_args
```

**Comment by rcombs** [comment]
Applied suggestions, verified, re-pushed.

File: Formula/libplacebo.rb [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  def install
+    python = "python3.10"
+    venv_root = buildpath/"venv"
+    venv = virtualenv_create(venv_root, python)
+    venv.pip_install resources
+    ENV.prepend_path "PYTHONPATH", venv_root/Language::Python.site_packages(python)
+
+    system "meson", "setup", "build",
+           "-Dvulkan-registry=#{Formula["vulkan-headers"].share}/vulkan/registry/vk.xml",
+           *std_meson_args
```
Comment (carlocab) [inline Formula/libplacebo.rb:43]:
```suggestion
    system "meson", "setup", "build",
                    "-Dvulkan-registry=#{Formula["vulkan-headers"].share}/vulkan/registry/vk.xml",
                    *std_meson_args
```

Comment (rcombs) [reply Formula/libplacebo.rb]: Done.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**